lib/shared/addon/components/cluster-driver/driver-rke/component.js

```diff
@@ -763,16 +763,16 @@ export default InputTextFile.extend(ManageLabels, ClusterDriver, {
       // get rid of undefined
       config = JSON.parse(JSON.stringify(config));
 
-      // We are assuming that the presentce of name indicates that
-      // cloud_provider should be present otherwise we remove it.
-      if ((!isEmpty(cpConfig) && !isEmpty(cpConfig.name)) || (this.isEdit && !isEmpty(cpConfig) && isEmpty(cpConfig.name))) {
-        delete config.rancher_kubernetes_engine_config.cloud_provider;
-      }
 
       if ( cpConfig ) {
         cpConfig = removeEmpty(cpConfig, EXCLUDED_KEYS);
-
-        if (cpConfig.azureCloudProvider) {
+        if (cpConfig.name === 'aws') {
+          config.rancher_kubernetes_engine_config.cloud_provider = {
```

Comment:
I believe this needs to be replaced with the following because `awsCloudProvider` has a couple non-empty params and the way your doing it resets the `awsCloudProvider` when you're editing and going to the yaml view. 
```
          config.rancher_kubernetes_engine_config.cloud_provider = {
            ...config.rancher_kubernetes_engine_config.cloud_provider,
            type:             'cloudProvider',
            awsCloudProvider: {
              ...config.rancher_kubernetes_engine_config.cloud_provider.awsCloudProvider,
              type: 'awsCloudProvider'
            },
          };
```

Reply:
There may be a better way to do this but we have to make sure we're not overriding the object that exists with just a type object.

Reply:
What if we instead did this instead?
`if (cpConfig.name === 'aws') && !cpConfig. awsCloudProvider`

We should only be adding the awsCloudProvider stuff if it doesn't exist.